src/providers/Participant/ParticipantProvider.tsx

```diff
@@ -90,21 +89,19 @@ const ParticipantProvider = ({ children }: Props) => {
   }, [projectModuleList]);
 
   useEffect(() => {
-    if (allProjectModuleList && allProjectModuleList.status === 'success' && allProjectModuleList.data) {
-      const allProjectModules = allProjectModuleList.data;
-      const nextModuleProjects = (participantProjects || []).filter(
-        (project) =>
-          allProjectModules.findIndex(({ id, modules }) => id === project.id && modules && modules.length > 0) !== -1
-      );
+    if (moduleProjectsList && moduleProjectsList.status === 'success' && moduleProjectsList.data) {
+      const nextModuleProjects = moduleProjectsList.data
```

Comment:
I wonder if you can satisfy all the data with the search query alone. This is fine how you have it, but it seems like we might be able to get everything in one shot.

Reply:
Should be able to! This is likely a backlog ticket for improving our API usages in general. Maybe we can start backlog of what API calls can be consolidated.